### ISSUE TITLE
PPO 

### ISSUE THREAD

**Issue by Plankson** [eleurent/highway-env#331]
Hello! I use stable_baseline PPO training in highway-fast-v0(continues), but can not get a good policy. The car keeps spinning and do not go forward. Is the hyperparameter setting problem?  And If you have test PPO in this environment, could you please give the hyperparameters of PPO?

**Comment by eleurent** [comment]
I think I have used the parameters from this script: https://github.com/eleurent/highway-env/blob/master/scripts/sb3_highway_ppo.py

**Comment by Plankson** [comment]
The highway-fast-v0 agent with continues action space I trained is tqc (that is , PPO training is still have the same problem discribed  in https://github.com/eleurent/highway-env/issues/268#issue-1141499378, so I follow him using tqc training), however the car is still wiggle slightly, I have no reason for why it can be like this. 

https://user-images.githubusercontent.com/26018849/175204515-1212a824-5f8f-4e37-b32e-8c32f1566e2a.mp4

Maybe I need to increase the reward weight of speed?(still have no reason about the wiggle : (

**Comment by eleurent** [comment]
> still have no reason about the wiggle 

I guess the problem is that it doesn't really have a reason NOT to wiggle, this behavior is not really penalized.

I haven't experimented a lot with continuous actions myself, but maybe you could edit the reward to penalize the amplitude of the steering action taken, or the yaw rate of the vehicle, to incentivize it to keep a constant heading as much as possible?

**Comment by victorfors** [comment]
Hi, I will share my settings which might be of help. I think the primary thing is that it might help with a higher policy frequency. I did use highway-v0 but think highway-fast-v0 should also work.

```
# Enviroment settings
env_kwargs = {
    'id': 'highway-v0', 
    'config': {
        "action": {
            "type": "ContinuousAction",
        },
        "policy_frequency": 5,
        "simulation_frequency": 10,
        "disable_collision_checks": True,
        "duration": 40*5, # This might have changed in newer releases of highway-env
        "vehicles_density": 1.0,
        "vehicles_count": 25,
    }
}

def make_configure_env(**kwargs):
    env = gym.make(kwargs["id"])
    env.configure(kwargs["config"])
    env.reset()
    return env

n_envs = 8
env = make_vec_env(make_configure_env, n_envs=n_envs, vec_env_cls=SubprocVecEnv, env_kwargs=env_kwargs)


model = TQC("MlpPolicy",
                        env,
                        policy_kwargs=dict(
                           n_critics=5,
                           n_quantiles=25,
                           net_arch=dict(pi=[256, 256], qf=[512, 512, 512]),
                           log_std_init=-3,
                           ),
                        batch_size=256,
                        learning_rate = 3e-4,
                        train_freq=8,
                        gradient_steps=8,
                        tau=0.005,
                        gamma=0.95,
                        verbose=2,
                        learning_starts=100,
                        use_sde=True,
                        tensorboard_log="highway_tqc/")
```

After 30,000 steps:
https://user-images.githubusercontent.com/99081590/177425917-3fb1492c-bcd6-4df2-a812-d26e43fabdce.mp4
After 300,000 steps:
https://user-images.githubusercontent.com/99081590/177426079-cc301df4-b3f1-47d2-911b-bc9ed51a059e.mp4
After 2,500,000 steps:
https://user-images.githubusercontent.com/99081590/177425920-8bc2cbde-9e98-4ce8-aa26-4e330f6c12f2.mp4

There is some very aggressive turning when training for longer which would probably need some modification of the reward function to get rid of.

**Comment by Plankson** [comment]
> Hi, I will share my settings which might be of help. I think the primary thing is that it might help with a higher policy frequency. I did use highway-v0 but think highway-fast-v0 should also work.

Thanks a lot!  I'll try it on highway-fast-v0  :-) 👍

**Comment by GGGQAQ** [comment]
Sorry, I can't watch the video anymore. May I ask if there is a trained model that I can use? I would like to do some simple tests.
